src/core/webview/ClineProvider.ts

```diff
@@ -484,12 +497,18 @@ export class ClineProvider implements vscode.WebviewViewProvider {
 							// await this.postStateToWebview() // new Cline instance will post state when it's ready. having this here sent an empty messages array to webview leading to virtuoso having to reload the entire list
 						}
 
+						break
+					case "alwaysAllowWrite":
+						await this.updateAlwaysAllowWrite(message.bool)
+						break
+					case "alwaysAllowExecute":
+						await this.updateAlwaysAllowExecute(message.bool)
 						break
 					// Add more switch case statements here as more webview message commands
 					// are created within the webview context (i.e. inside media/main.js)
 				}
 			},
-			null,
+			undefined,
```

Comment:
**Question:** Why did this change from `null,` to `undefined,`?

In TypeScript's `strictNullChecks` mode, `null` and `undefined` are treated as distinct types and aren't interchangeable.

**suggestion:** revert this from `undefined,` back to `null,`
```suggestion
			null,
```